### ISSUE TITLE
Any function with subtraction is given an infinite log prior

### ISSUE THREAD

**Issue by tomasdsousa** [DeaglanBartlett/katz#5]
`from katz.prior import KatzPrior

n = 2
basis_functions = [["a", "x"],
        ["inv","arcsin","sqrt", "exp", "log", "sin", "cos", "tanh", "square", "cube"],
        ["+", "-", "*", "/" , "pow"]]

kp = KatzPrior(n, basis_functions, 'katz/data/FeynmanEquations.csv', 'katz/data/NewFeynman.csv')

for eq in ['a0 + x',
        ['+','a0','x'],
        'a0 - x',
        ['-','a0','x'] ]:
    p = kp.logprior(eq)
    print(eq, p)`

This example code will return a finite log prior for the string representation of a0 - x, but -infinity for its tree representation.

**Comment by DeaglanBartlett** [comment]
Running
```
for eq in ['a0+x', 'a0-x']:
    expr, nodes, c = generator.string_to_node(eq, kp.coder.basis_functions, locs=locs, evalf=True, kernS_only=True)
    print(expr)
    print(nodes.to_list(kp.coder.basis_functions))
```
gives
```
a0 + x
['Add', 'a0', 'x']
a0 - x
['Add', 'a0', 'Mul', '-1', 'x']
```
i.e. the `string_to_node` function gives a tree for `a0-x` which doesn't contain a minus sign, but interprets `a0-x` as `a0+(-1)*x`. Ideally we would like this to be passed as `a0-x` This is why the '-' operator never appears in the corpus and hence the logprior value is non-finite, since any '-' in the corpus is being read as `+(-1)*`. So to rectify this issue, we need to ensure that any instance of `+(-1)*` is replaced with just a -.

**Comment by DeaglanBartlett** [comment]
This has been fixed with PR #6. This also required the ESR code to be updated so that the `DecoratedNode` class can identify this pattern of operators when producing a list of operators which defines the tree. So please update both the `katz` and `esr` code to run this.
